PSI_Tracker.js

```diff
@@ -0,0 +1,645 @@
+    if(api == "Sustainability"){
+      fields = fields.filter(d=>d.method == "Sustainability")
+      green_domains = SpreadsheetApp
+                        .getActiveSpreadsheet()
+                        .getSheetByName("Green Domains (GWF)")
+                        .getRange("F2:J")
+                        .getValues()
+                        .filter(d=>d[4]==true)
+                        .map(d=>d[0])
+      Logger.log("GREEN DOMAINS")
+      Logger.log(green_domains)  
+    }
+
+
+    // STEP 2: We iterate through all the fields
+    for (let i = 0; i < fetch.length; i++) {
+        let fields_for_debug = [new Date().toISOString().slice(0, 10), batch[i].label, batch[i].url, batch[i].device, batch[i].origin];
+        let debug = "";
+        // Indicate that we worked on that row in the Config tab
+        sheet.getRange(batch[i].id+1,COLUMN_STATUS+1).setValue("✅")
+        // Read and parse the data for that specific cakk
+        const data = fetch[i];
+        Logger.log(data)
+        let content = {};
+        // Try to parse the answer from the API
+        try{
+          content = JSON.parse(data);
+        }
+        // If an error occurs (ex: HTML returned instead of JSON), throw it here
+        catch(e){
+          sheet.getRange(batch[i].id+1,COLUMN_STATUS+1).setValue("🟥")
+          sheet.getRange(batch[i].id+1,COLUMN_STATUS+1).setNote("Error " + e.toString() + "\n\nFull error:\n" + JSON.stringify(data));
+          debug += "🟥 " + "Error " + e.toString() + "\n\nFull error:\n" + JSON.stringify(data) + " | " 
+          fields_for_debug.push(debug)
+          DEBUG_SHEET.appendRow(fields_for_debug);
+        }
+        Logger.log(content)
+        // If an error occured, flagged it in the Config tab, under the Status
+        if(content["error"]){
+            sheet.getRange(batch[i].id+1,COLUMN_STATUS+1).setValue("❌")
+            sheet.getRange(batch[i].id+1,COLUMN_STATUS+1).setNote("Error " + JSON.stringify(content["error"]))
+            debug += "❌ " + "Error " + JSON.stringify(content["error"]) + " | " 
+            fields_for_debug.push(debug)
+            DEBUG_SHEET.appendRow(fields_for_debug);
+        }
+        else{
+            // Build a dic that will contain all the extracted information
+            let output = {
+              "Date": new Date().toISOString().slice(0, 10),
+              "Label": batch[i].label,
+              "URL": batch[i].url,
+              "Device": batch[i].device,
+              "URL / Origin": batch[i].origin
+            };
+            // Iterate through all the fields and eval the function to get their respective values, adding them to the dic
+            for (let j = 0; j < fields.length; j++) {
+                try{
+                    // Origin field is used in some calls to know if there was no URL-level data and we reverted to Origin level
+                    let origin = batch[i].origin;
+                    let green_domains_ = green_domains;
```

Comment:
I can't find where `green_domains_` (with the trailing underscore) is used again, but the name is hard to tell from the `green_domains` (no trailing underscore). Is there a better name that could be used here?

Reply:
yes, here it is to surface the variable for the `eval` section, and be sure it is accessible for the GWF, will try to add more comment on this!